core/camel-api/src/main/java/org/apache/camel/spi/DataFormat.java

```diff
@@ -54,4 +54,13 @@
      * @throws Exception can be thrown
      */
     Object unmarshal(Exchange exchange, InputStream stream) throws Exception;
+
+    /**
+     * Indicates whether {@code null} is allowed as value of a body to unmarshall.
+     *
+     * @return {@code true} if {@code null} is allowed, {@code false} otherwise.
+     */
+    default boolean allowNullBody() {
```

Comment:
Is this correct? It should be specific to Jackson and not generic to data format.

I'm from mobile so I might losing something

Reply:
That's a good remark, actually, the failure doesn't happen in the Jackson `DataFormat` itself but in the `UnmarshalProcessor`.  Indeed, The failure is due to the fact that it cannot find a converter to convert a `null` value into an `InputStream` which occurs when calling `in.getMandatoryBody(InputStream.class)`. That is why I had to manage it at a higher level, moreover @davsclaus proposed in the same ticket to eventually have it for all the existing DataFormat which could give more sense to this approach. Do you see a better/simpler approach? If I need to apply the same logic to the other DataFormats, I had better to have the best approach.

Reply:
Well, I didn't read the suggestion from @davsclaus, in that case it makes completely sense as a basic step to then provide the feature on all the dataformat. So fine with me. Thanks for the explanation.

Reply:
Yeah I can't think of a better way as there is no common DataFormatConfiguration class that they all extend etc where we could have common options. 

But this is really a useful feature to let the data format not fail on null bodies.

Especially for use-cases with json where we would often use jackson.